File: dart/funciones/6.dart
import 'dart:io';
import 'dart:math';

void factorial() {
    
    print("Digite un número NO negativo.");
    
    int num = int.parse(stdin.readLineSync()!);
    
    int factorial = 1;
    
    for(int i = 2; i<=num; i++){
        
        factorial *= i;
        
    }
  
  print("${factorial}");
  
}


void main(){
    
    factorial();
   
}

File: funciones/6.php
<?php

$num = readline("Digite un número entero NO negativo: ");

$num = intval($num); //que sea entero

function factorial($n) {

    if($n <= 1){

        return 1;

    }else{

        return $n * factorial($n - 1);

    }

}

$resultado = factorial($num);

echo "El factorial de tu número es igual a: $resultado";

?>